src/open_inwoner/templates/django/forms/widgets/select.html

```diff
@@ -0,0 +1,5 @@
+<select name="{{ widget.name }}" {% if widget.value.0 == "" %}class="{{widget.attrs.class}} muted"{% endif %}{% include "django/forms/widgets/attrs.html" %}>{% for group_name, group_choices, group_index in widget.optgroups %}{% if group_name %}
```

Comment:
It doesn't seem to be possible to style a `<select>` based on the selected `<option>` using CSS and I didn't want to create a custom widget that would have to be set on each instance of `...ChoiceField`, so I opted for this approach instead

Reply:
Note: We put all our Form Component templates in this project in this particular directory here: _src/open_inwoner/components/templates/components/Form_
So I don't think we should put anything into the Django templates Dir.

Also: with this approach we need to create a 1st option inside all Selects that should start with some kind of empty default, like here:
http://127.0.0.1:8000/mijn-uitkeringen/  
So I guess you should add one there :-) 
Not sure what to call it, perhaps "Select month".
Same for:
http://127.0.0.1:8000/mijn-uitkeringen/jaaropgaven/ "Select year"

More pages with selects, that are probably allright: 
http://127.0.0.1:8000/mijn-profiel/contacts/
http://127.0.0.1:8000/mijn-profiel/actions/
And ">> Mijn profiel >> Mijn acties >> Bewerk [Actie title]"

<img width="1317" alt="benefits-select" src="https://github.com/maykinmedia/open-inwoner/assets/118177951/e7b5fe21-4cdb-4d1b-b0f3-1b158ad46190">

Reply:
> Also: with this approach we need to create a 1st option inside all Selects that should start with some kind of empty default, like here:

Just to be sure: do you mean that having `Sep 2023` **not** muted as the initial value would be okay as well? I could modify the widget to check if the initial value is empty or not and apply the class with the muted styling based on that

> More pages with selects, that are probably allright:
http://127.0.0.1:8000/mijn-profiel/contacts/

I checked the contacts page, the change I mentioned above would cause "Alle" to be displayed muted, because it's value is `""`. Is this correct?
![image](https://github.com/maykinmedia/open-inwoner/assets/29249171/5d84cee1-d5f7-4441-96a2-4fa759edbfd1)

> Note: We put all our Form Component templates in this project in this particular directory here: src/open_inwoner/components/templates/components/Form
So I don't think we should put anything into the Django templates Dir.

I put the template in `templates/django/forms/widgets/select.html` to override the default widget that Django uses under the hood, which makes this change available for all selects without having to manually add classes to each occurring select. I'm not sure if there is another way to override a default Django widget without having to monkey patch stuff, which seems like it would be unnecessary in this case in my opinion.

If it turns out we do not want this styling applied to each select or if there is no generic solution, the only solution is probably to add the muted class to all selects manually (which could be kind of a chore, I'm not sure how many selects there are :stuck_out_tongue:)

Reply:
Indeed: the first option in the selects in the Maandopgaven/Jaaropgaven should **not** be muted because it's already showing a real option - the muted style is a UX indication, so should only be applied to a default option to indicate to the user something can be selected/filtered.
I'm not sure anymore but I tried to tackle this with **CSS-only** a couple of months ago and I think I found that some selects weren't using the value attribute at all, which made them empty, which meant that too many of them had a zero value so it would show the mute style too often.

In my previous answer I posted all of the places that have a Select; so currently i'd think there's only a maximum of 6 selects in the entire project.
And yes: I'd say 'alle' for Contacts should be muted because it means nothing has been filtered by the  select yet.

As for the location of the templates and if we just simply wish to override: I am not sure - I'd have to ask @alextreme or @Bartvaderkin.

Reply:
> In my previous answer I posted all of the places that have a Select; so currently i'd think there's only a maximum of 6 selects in the entire project.

Ah alright, in that case I think all of them are displayed correctly:

![image](https://github.com/maykinmedia/open-inwoner/assets/29249171/d378dec4-e7ee-432d-971a-848a03d45430)

![image](https://github.com/maykinmedia/open-inwoner/assets/29249171/7ce8e39d-98a1-4787-91e1-64d2c7d25f97)

![image](https://github.com/maykinmedia/open-inwoner/assets/29249171/ba7f4544-a5ce-4825-961c-2735da9975bf)

![image](https://github.com/maykinmedia/open-inwoner/assets/29249171/d086bf48-22e3-48e0-9bd3-fe0b3c225c14)

![image](https://github.com/maykinmedia/open-inwoner/assets/29249171/74822ddb-67eb-4e94-8164-70927ca78de9)

Reply:
If that's true then i will approve this :-) for new selects we'll need to make a correct model with values I guess.